src/BarChart/BarSeries/BarTargetMarker.tsx

```diff
@@ -0,0 +1,220 @@
+
+  /**
+   * Color of the target marker line.
+   */
+  fill: string;
+
+  /**
+   * Color of the positive delta line.
+   */
+  positiveDeltaFill: string;
+
+  /**
+   * Color of the negative delta line.
+   */
+  negativeDeltaFill: string;
+
+  /**
+   * Number of the bars in the bar group. Set internally by `BarSeries`.
+   */
+  barCount: number;
+
+  /**
+   * Layout of bar chart to render. Set internally by `BarSeries`.
+   */
+  layout: Direction;
+
+  /**
+   * Whether to animate the enter/update/exit. Set internally by `BarSeries`.
+   */
+  animated: boolean;
+
+  /**
+   * Class name to apply to the text.
+   */
+  className?: any;
+
+  /**
+   * Thickness of the line.
+   */
+  lineThickness?: number;
+
+  /**
+   * Thickness of the target marker line.
+   */
+  targetLineThickness?: number;
+
+  /**
+   * Thickness of the difference/delta line that shows distance between target and actual value.
+   */
+  deltaLineThickness?: number;
+}
+
+export const BarTargetMarker: FC<Partial<BarTargetMarkerProps>> = ({
+  data,
+  height,
+  width,
+  x,
+  y,
+  animated,
+  layout = 'vertical',
+  scale,
+  index,
+  barCount,
+  fill = '#fff',
+  positiveDeltaFill = '#00C49F',
+  negativeDeltaFill = '#FF7361',
+  targetLineThickness = 2,
+  deltaLineThickness = 1
+}) => {
+  const isVertical = useMemo(() => layout === 'vertical', [layout]);
+  const valuePosition = useMemo(() => scale(data.value), [data.value, scale]);
+  const targetPosition = useMemo(
+    () => scale(data.target),
+    [data.target, scale]
+  );
+
+  const enterProps = useMemo(() => {
+    let newY = y;
+    let newX = x;
+
+    if (isVertical) {
+      newY = targetPosition;
+    } else {
+      newX = targetPosition;
+    }
+
+    return {
+      translateX: newX,
+      translateY: newY,
+      opacity: 1
+    };
+  }, [isVertical, targetPosition, x, y]);
+
+  const exitProps = useMemo(() => {
+    let newY = y;
+    let newX = x;
+
+    if (isVertical) {
+      const maxY = Math.max(...scale.range());
+      newY = maxY;
+
+      newX = newX + width / 2;
+    } else {
+      const minX = Math.min(...scale.range());
+      newX = minX;
+    }
+
+    return {
+      translateY: newY,
+      translateX: newX,
+      opacity: 0
+    };
+  }, [isVertical, scale, width, x, y]);
+
+  const delay = useMemo(() => {
+    let delay = 0;
+    if (animated) {
+      return (index / barCount) * 0.5;
+    }
+
+    return delay;
+  }, [animated, barCount, index]);
+
+  const delta = useMemo(() => {
+    return Math.abs(valuePosition - targetPosition);
+  }, [valuePosition, targetPosition]);
+
+  const isTargetGreaterThanValue = useMemo(() => {
+    return targetPosition > valuePosition;
+  }, [valuePosition, targetPosition]);
+
+  return (
+    <motion.g
+      initial={exitProps}
+      animate={enterProps}
+      exit={exitProps}
+      transition={{
+        ...DEFAULT_TRANSITION,
+        delay
+      }}
+    >
+      {isVertical && (
+        <motion.rect
```

Comment:
Seems like these two could be combined w/ some smarter prop passing.

Reply:
I split this into 2 elements to simplify the logic. If using 1 element it to hard to read.

Reply:
This reads cleaner to me because i can easily scan and see how things are mapped rather than look in 2 places.

```
const height = isVertical ? whatever : whatever;
const fill = '';
const animate = {}
```

Reply:
you already do it here too lol - https://github.com/reaviz/reaviz/pull/248/files#diff-08f65d895191f47cce4bd6a1957428d1c734980b4529233ca86d3d1b32d83b85R172